base-server.d.ts

```diff
@@ -0,0 +1,759 @@
+import { Log, Meta, Store, ServerConnection, TestTime } from '@logux/core'
+import { Server as HTTPServer } from 'http'
+
+import { Context, ChannelContext, LoguxPatternParams } from './context'
+import { ServerClient } from './server-client'
+
+export type LoguxMeta = Meta & {
+  /**
+   * Action processing status
+   */
+  status?: 'waiting' | 'processed' | 'error'
+
+  /**
+   * Node ID of the server received the action.
+   */
+  server: string
+
+  /**
+   * Array and channel string: all clients subscribed
+   * to listed channels will receive the action.
+   */
+  channels?: string[]
+
+  /**
+   * Array and user string: all clients with listed user
+   * IDs will receive the action.
+   */
+  users?: string[]
+
+  /**
+   * Array and client string: all clients with listed client
+   * IDs will receive the action.
+   */
+  clients?: string[]
+
+  /**
+   * Array and node string: all clients with listed node
+   * IDs will receive the action.
+   */
+  nodes?: string[]
+}
+
+export type LoguxUserAction = {
+  type: string
+}
+
+export type LoguxSubscribeAction = {
+  type: 'logux/subscribe'
+  channel?: string
+  since?: {
+    id: string
+    time: number
+  }
+}
+
+export type LoguxUnsubscribeAction = {
+  type: 'logux/unsubscribe'
+  channel?: string
+}
+
+export type LoguxProcessedAction = {
+  type: 'logux/processed'
+  id?: string
```

Comment:
We have special `ID` type in `@logux/core` (to be sure that ID was copied from the `meta.id`)